feat: Add clear button to search bar

techwithanirudh opened the pull request:
This commit introduces a clear button to the SearchBar component using the X icon from Lucide-React. When the user enters a query in the input field, the clear button appears allowing them to easily remove the search term. The clear button is hidden when there is no search term entered.

techwithanirudh committed:
feat: Add clear button to search bar

This commit introduces a clear button to the SearchBar component using the X icon from Lucide-React. When the user enters a query in the input field, the clear button appears allowing them to easily remove the search term. The clear button is hidden when there is no search term entered.

techwithanirudh committed:
Refactor SearchBar component to improve user experience

Changed SearchBar's input field to add padding on the left side and an absolute positioned search icon. Also, added absolute positioned X icon on the right side when there is an input value, ensuring a better user experience.

techwithanirudh committed:
Merge branch 'main' into searchbar

ClaraLeigh commented:
Hey, I love the enthusiasm but I'm not sure this feature will help our end User Experience.
 
1. This will create content shake, when the content is empty it will not have an icon, where as soon as you start typing the ux will change. In industry this is generally frowned upon these days and it is better to just make something look inactive.
2. I'm not sure we need this feature, everyone knows how to delete and with its position someone might click on it when going for the send button
3. On mobile you'd loose space and there is already a lack of space when trying to type things

techwithanirudh committed:
Refactor SearchBar component to show Clear Search icon dynamically

This commit makes changes to the SearchBar React component to render the Clear Search X icon only when the input field has a value. A showClearIcon state using useState hook is added and updated every time the input value changes. The useEffect hook is used to handle the case when the user clears the input value. This allows better UX by providing clear intent to the user that the icon is clickable and will clear the search query.

techwithanirudh committed:
Merge branch 'searchbar' of https://github.com/techwithanirudh/chatgpt-clone into searchbar

techwithanirudh committed:
Merge branch 'main' into searchbar

techwithanirudh committed:
Improve UX: Add styling to clear button & export button

This commit modifies the NavLinks component to improve user experience by removing a rounded styling to the "Clear conversations" and "Export conversations" buttons. Prior to this change, the buttons had a rounded styling.

techwithanirudh committed:
Refactor submit button styling for improved accessibility and readability.

Changed submit button styling for better accessibility and readability, including adjustments to padding and hover effects. The new styles ensure that the button is easily clickable for all users, while also improving its visual appearance.

techwithanirudh committed:
hotfix

ClaraLeigh commented on client/src/components/Nav/NavLinks.jsx:
```diff
@@ -77,7 +77,7 @@ export default function NavLinks({ clearSearch, isSearchEnabled }) {
                   <NavLink
                     className={cn(
-                      'flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700',
+                      'flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700 rounded-none',
```
do we need the rounded-none class here? it wasn't there previously

techwithanirudh replied:
Yes, dannya made some changes to the css that kinda degraded the similarity between a few menu items, so I fixed them

ClaraLeigh commented on client/src/components/Nav/SearchBar.jsx:
```diff
@@ -37,8 +48,15 @@ const SearchBar = forwardRef((props, ref) => {
+      <X
+        className={`h-5 w-5 absolute right-3 cursor-pointer ${showClearIcon ? 'opacity-100' : 'opacity-0'} transition-opacity duration-1000`}
+        onClick={() => {
+          setSearchQuery('');
+          clearSearch();
```
Do these lines do the same thing? also I'm not sure I can see a clearSearch() function included on this script anywhere?

techwithanirudh replied:
I don't think so, dannya used it in ```handleKeyUp``` so I thought that was the best way to clear the input

ClaraLeigh replied:
ah I see the function included above now, just wanted to check it was linking somewhere

ClaraLeigh commented on client/src/components/Nav/NavLinks.jsx:
```diff
@@ -91,7 +91,7 @@ export default function NavLinks({ clearSearch, isSearchEnabled }) {
                 <Menu.Item as="div">
                   <NavLink
-                    className="flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700"
+                    className="flex w-full cursor-pointer items-center gap-3 px-3 py-3 text-sm text-white transition-colors duration-200 hover:bg-gray-700 rounded-none"
```
same again: do we need the rounded-none class here? it wasn't there previously

techwithanirudh replied:
Yes, dannya made some changes to the css that kinda degraded the similarity between a few menu items, so I fixed them

ClaraLeigh commented on client/src/components/Input/SubmitButton.jsx:
```diff
@@ -88,7 +88,7 @@ export default function SubmitButton({
         className="group absolute bottom-0 right-0 flex h-[100%] w-[50px] items-center justify-center bg-transparent p-1 text-gray-500"
       >
-        <div className="m-1 mr-0 rounded-md p-2 pb-[10px] pt-[10px] group-hover:bg-gray-100 group-disabled:hover:bg-transparent dark:group-hover:bg-gray-900 dark:group-hover:text-gray-400 dark:group-disabled:hover:bg-transparent">
+        <div className="m-1 mr-0 rounded-md pt-[11px] pb-[9px] pl-[9.5px] pr-[7px] group-hover:bg-gray-100 group-disabled:hover:bg-transparent dark:group-hover:bg-gray-900 dark:group-hover:text-gray-400 dark:group-disabled:hover:bg-transparent">
```
not sure I understand the changes to padding here? whats the reasoning?

techwithanirudh replied:
The padding was off for the submit button, so I fixed this. This is a overall css improvement P.R

ClaraLeigh reviewed (changes requested):
Also I saw you posted the screenshot in discord. Might be good to pop it in here too so others can see the visual change. Before I saw that screenshot I actually thought we were talking about a different area

ClaraLeigh reviewed (approved):
Lgtm. Assuming the rounded-none css is needed but @danny-avila would know for sure as you mention its a bug from their css changes. Same with the extra submit button padding

techwithanirudh committed:
Improve UI styling in Conversation component

Changed the background color and hover effect of the conversation link in Conversation component to make it more visually appealing. The previous background color was '#2A2B32' and now it's 'gray-800'. The 'px-4' class has also been changed to 'hover:pr-4' for better readability.

techwithanirudh committed:
Merge branch 'main' into searchbar

techwithanirudh committed:
Merge branch 'main' into searchbar

techwithanirudh committed:
Merge branch 'main' into searchbar

techwithanirudh committed:
Merge branch 'main' into searchbar

danny-avila committed:
Merge branch 'main' into searchbar

danny-avila committed:
Merge branch 'main' into searchbar
